subprojects/build-plugins/src/main/kotlin/org/gradle/kotlin/dsl/build/plugins/GeneratedSourcesLayoutPlugin.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.build.plugins
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.file.Directory
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.SourceSet
+
+import org.gradle.kotlin.dsl.*
+
+import javax.inject.Inject
+
+
+internal
```

Comment:
Should be open, right?
Or is internal "similar" to open?

Reply:
It's not similar. `internal` is "similar" to package private in Java.

Reply:
Then it should be open, right? 😂 
Otherwise Gradle can't create the Proxy-Class of this Plugin...

Reply:
Plugin classes don't need to be open.